File: mobile/lib/features/main/data/mappers/school_mapper.dart
import 'package:mobile/features/main/main.dart';

class SchoolMapper {
  static School schoolJsonToEntity(Map<String, dynamic> json) => School(
    id: json['id'],
    name: json['name'],
    logoImage: json['logo_image'],
    type: json['type'],
    phone: json['phone'],
    role: json['role'],
  );
}

File: web/features/school/data/mappers/school/school.mapper.ts
import type { School } from "../../../domain/entities/school";
import type { SchoolResponseDto } from "../../schemas/school-response.schema";

export function toSchoolEntity(dto: SchoolResponseDto): School {
  return {
    id: dto.id,
    name: dto.name,
    logoImage: dto.logo_image,
    type: dto.type,
    phone: dto.phone,
    role: dto.role,
  };
}

export function toSchoolEntityList(dtos: SchoolResponseDto[]): School[] {
  return dtos.map(toSchoolEntity);
}
